src/Services/Qdm/Services/AbstractObservationService.php

```diff
@@ -63,8 +69,13 @@ public function makeQdmModel(QdmRecord $recordObj)
         $qdmModel = new $modelClass([
             '_id' => $id,
             'id' => $id,
-            'relevantDatetime' => new DateTime([
-                'date' => $record['date']
+            'relevantPeriod' => new Interval([
+                'low' => new DateTime([
+                    'date' => $record['date']
+                ]),
```

Comment:
The test ask for relevantPeriod in some measures and relevantDateTime in others. So both need to work depending on how you want to report. However this was a known issue reported in the 2021 reporting period so interpret how you want.

Guidance for reporting eCQMs with QDM data types allowing 
either relevantDatetime or relevantPeriod attributes for the 2021 
reporting/performance period
For all eCQMs where QDM data types support both relevantDatetime and 
relevantPeriod attributes, the Cypress, QualityNet, and The Joint Commission 
systems will allow submitters to submit data either as relevantDatetime 
(effectiveTime/@value) or relevantPeriod (effectiveTime/low and 
effectiveTime/high) in a QRDA Category I file.
For example, for CMS129v10, even though the measure specification uses 
relevantPeriod for ["Procedure, Performed": "Prostate Cancer Treatment"], 
submitters may submit the timing of this data element either as relevantDatetime 
or relevantPeriod in QRDA Category I.
For purposes of hospital measure calculation:
1. If a relevantDatetime is reported, it will be converted to a relevantPeriod.
2. When a relevantPeriod is reported and a relevantDatetime is needed for evaluation, 
the low value will be used unless not provided, then the high time will be used.

Reply:
Yes, @stephenwaite you need to have both relevantPeriod and relevantDatetime, sorry for the confusion

Reply:
the `relevant_date_period_or_null_flavor` function in the date helper returns either I think based on what's sent

Reply:
so should be all good now

![Screenshot from 2022-05-19 13-47-36](https://user-images.githubusercontent.com/5618372/169364841-c33b9ed7-fa1f-4d32-8bf8-0854e71eb248.png)

Reply:
oh, I get what you're saying sorry